app/models/email_address.rb

```diff
@@ -33,6 +33,15 @@ def confirmation_period_expired?
     Time.zone.now > expiration_time
   end
 
+  def domain
+    email&.split('@')&.last
+  end
+
+  def is_fed_email?
+    return false unless domain
+    FedEmailDomains.email_is_fed_domain?(domain)
```

Comment:
I like the style of avoiding `is_` in predicate methods [(styleguide link)](https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#bool-methods-prefix)
```suggestion
  def fed_email?
    return false unless domain
    FedEmailDomains.email_is_fed_domain?(domain)
```